src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/ProjectSystem/FallbackProjectManager.cs

```diff
@@ -0,0 +1,163 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Composition;
+using System.IO;
+using Microsoft.CodeAnalysis.Razor.Workspaces;
+
+namespace Microsoft.CodeAnalysis.Razor.ProjectSystem;
+
+/// <summary>
+/// This class is responsible for maintaining project information for projects that don't
+/// use the Razor or Web SDK, or otherwise don't get picked up by our CPS bits, but have
+/// .razor or .cshtml files regardless.
```

Comment:
Do we need CPS to tell us about Razor projects at all? Or, could this just be the way that we learn about .razor files.

Reply:
In general, I think using Roslyn over CPS is the way forward but for two caveats:

1) We still need CPS now because there are msbuild knobs that can be controlled which Roslyn does not expose. Whether these knobs are used or important in 2023 is an open question, and it isn't impossible to create a system to allow Roslyn to expose the info anyway.
	1.1) Note in VS Code we completely ignore these extra bits of data and it hasn't bitten us _yet_
2) For these fallback projects in particular, Roslyn doesn't even know about the documents, as they're not `AdditionalItems` in msbuild, so this particular approach is not the way to go.